kornia/augmentation/augmentations.py

```diff
@@ -124,3 +174,102 @@ def random_hflip(input: torch.Tensor, p: float = 0.5, return_transform: bool = F
+                 contrast: FloatUnionType = 0., saturation: FloatUnionType = 0.,
+                 hue: FloatUnionType = 0., return_transform: bool = False) -> UnionType:
+    r"""Random color jiter of an image or batch of images.
+
+    See :class:`~kornia.augmentation.ColorJitter` for details.
+    """
+
+    def _input_check(factor: FloatUnionType, name: str, center: float = 0.,
+                     bounds: Tuple[float, float] = (0, float('inf'))):
+        r"""Check inputs and compute the corresponding factor bounds
+        """
+
+        if isinstance(factor, float):
+
+            if factor < 0:
+                raise ValueError(f"If {name} is a single number number, it must be non negative. Got {factor}")
+
+            factor_bound = torch.tensor([center - factor, center + factor])
+            factor_bound = torch.clamp(factor_bound, bounds[0], bounds[1])
+
+        elif (isinstance(factor, torch.Tensor) and factor.dim() == 0):
+
+            if factor < 0:
+                raise ValueError(f"If {name} is a single number number, it must be non negative. Got {factor}")
+
+            factor_bound = torch.tensor([torch.tensor(center) - factor, torch.tensor(center) + factor])
+            factor_bound = torch.clamp(factor_bound, bounds[0], bounds[1])
+
+        elif isinstance(factor, (tuple, list)) and len(factor) == 2:
+
+            if not bounds[0] <= factor[0] <= factor[1] <= bounds[1]:
+                raise ValueError(f"{name}[0] should be smaller than {name}[1] got {factor}")
+
+            factor_bound = torch.tensor(factor)
+
+        elif isinstance(factor, torch.Tensor) and factor.shape[0] == 2 and factor.dim() == 1:
+
+            if not bounds[0] <= factor[0] <= factor[1] <= bounds[1]:
+                raise ValueError(f"{name}[0] should be smaller than {name}[1] got {factor}")
+
+            factor_bound = factor
+
+        else:
+
+            raise TypeError(
+                f"The {name} should be a float number or a tuple with length 2 whose values move between {bounds}.")
+
+        return factor_bound
+
+    if not torch.is_tensor(input):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+
+    if not isinstance(return_transform, bool):
+        raise TypeError(f"The return_transform flag must be a bool. Got {type(return_transform)}")
+
+    brightness_bound = _input_check(brightness, 'brightness', bounds=(float('-inf'), float('inf')))
+    contrast_bound = _input_check(contrast, 'contrast', center=1.)
+    saturation_bound = _input_check(saturation, 'saturation', center=1.)
+    hue_bound = _input_check(hue, 'hue', bounds=(-.5, .5))
+
+    device: torch.device = input.device
+    dtype: torch.dtype = input.dtype
+
+    input = input.unsqueeze(0)
+    input = input.view((-1, (*input.shape[-3:])))
+
+    brightness_distribution = Uniform(brightness_bound[0], brightness_bound[1])
+    brightness_factor = brightness_distribution.rsample([input.shape[0]])
```

Comment:
in order to make the gpu integration tests easy -> `factor_bound.to(input.device).to(input.dtype)`

same for the rest

Reply:
Corrected in the following commit